File: bangunDatar.cpp
#include <iostream>
#include <cmath>

class BangunDatar {
public:
    virtual double hitungLuas() const {
        return 0.0;
    }
};

class Segitiga : public BangunDatar {
private:
    double alas;
    double tinggi;

public:
    Segitiga(double a, double t) : alas(a), tinggi(t) {}

    double hitungLuas() const override {
        return 0.5 * alas * tinggi;
    }
};

class Persegi : public BangunDatar {
private:
    double sisi;

public:
    Persegi(double s) : sisi(s) {}

    double hitungLuas() const override {
        return sisi * sisi;
    }
};

class Lingkaran : public BangunDatar {
private:
    double jari_jari;

public:
    Lingkaran(double r) : jari_jari(r) {}

    double hitungLuas() const override {
        return 3.14159265359 * jari_jari * jari_jari;
    }
};

int main() {
    int pilihan;
    double alas, tinggi, sisi, jari_jari;

    std::cout << "Program Menghitung Luas Bangun Datar" << std::endl;
    std::cout << "Pilih bentuk (1: Segitiga, 2: Persegi, 3: Lingkaran): ";
    std::cin >> pilihan;

    if (pilihan == 1) {
        std::cout << "Masukkan alas segitiga: ";
        std::cin >> alas;
        std::cout << "Masukkan tinggi segitiga: ";
        std::cin >> tinggi;
        BangunDatar* bentuk = new Segitiga(alas, tinggi);
        std::cout << "Luas Segitiga: " << bentuk->hitungLuas() << std::endl;
        delete bentuk;
    } else if (pilihan == 2) {
        std::cout << "Masukkan panjang sisi persegi: ";
        std::cin >> sisi;
        BangunDatar* bentuk = new Persegi(sisi);
        std::cout << "Luas Persegi: " << bentuk->hitungLuas() << std::endl;
        delete bentuk;
    } else if (pilihan == 3) {
        std::cout << "Masukkan jari-jari lingkaran: ";
        std::cin >> jari_jari;
        BangunDatar* bentuk = new Lingkaran(jari_jari);
        std::cout << "Luas Lingkaran: " << bentuk->hitungLuas() << std::endl;
        delete bentuk;
    } else {
        std::cout << "Pilihan tidak valid." << std::endl;
    }

    return 0;
}

File: bangunDatar.php
<!DOCTYPE html>
<html>
<head>
    <title>Kalkulator Luas Bangun Datar</title>
</head>
<body>
    <h2>Kalkulator Luas Bangun Datar</h2>
    
    <?php
    class BangunDatar {
        public function hitungLuas() {
            return 0.0;
        }
    }

    class Segitiga extends BangunDatar {
        private $alas;
        private $tinggi;

        public function __construct($alas, $tinggi) {
            $this->alas = $alas;
            $this->tinggi = $tinggi;
        }

        public function hitungLuas() {
            return 0.5 * $this->alas * $this->tinggi;
        }
    }

    class Persegi extends BangunDatar {
        private $sisi;

        public function __construct($sisi) {
            $this->sisi = $sisi;
        }

        public function hitungLuas() {
            return $this->sisi * $this->sisi;
        }
    }

    class Lingkaran extends BangunDatar {
        private $jari_jari;

        public function __construct($jari_jari) {
            $this->jari_jari = $jari_jari;
        }

        public function hitungLuas() {
            return M_PI * $this->jari_jari * $this->jari_jari;
        }
    }

    if ($_SERVER["REQUEST_METHOD"] == "POST") {
        $bentuk = $_POST["bentuk"];
        $luas = 0.0;

        if ($bentuk == "segitiga") {
            $alas = $_POST["alas"];
            $tinggi = $_POST["tinggi"];
            $segitiga = new Segitiga($alas, $tinggi);
            $luas = $segitiga->hitungLuas();
        } elseif ($bentuk == "persegi") {
            $sisi = $_POST["sisi"];
            $persegi = new Persegi($sisi);
            $luas = $persegi->hitungLuas();
        } elseif ($bentuk == "lingkaran") {
            $jari_jari = $_POST["jari_jari"];
            $lingkaran = new Lingkaran($jari_jari);
            $luas = $lingkaran->hitungLuas();
        }

        echo "Luas $bentuk: $luas";
    }
    ?>

    <form method="post" action="<?php echo htmlspecialchars($_SERVER["PHP_SELF"]); ?>">
        <label>Pilih Bentuk:</label>
        <select name="bentuk">
            <option value="segitiga">Segitiga</option>
            <option value="persegi">Persegi</option>
            <option value="lingkaran">Lingkaran</option>
        </select>

        <br><br>

        <label for="alas">Alas (untuk segitiga):</label>
        <input type="text" name="alas" id="alas">
        
        <label for="tinggi">Tinggi (untuk segitiga):</label>
        <input type="text" name="tinggi" id="tinggi">

        <label for="sisi">Sisi (untuk persegi):</label>
        <input type="text" name="sisi" id="sisi">

        <label for="jari_jari">Jari-jari (untuk lingkaran):</label>
        <input type="text" name="jari_jari" id="jari_jari">

        <br><br>

        <input type="submit" value="Hitung">
    </form>
</body>
</html>
